### ISSUE TITLE
What are the timeouts (read/write on the HTTP server on the ingress) & calls to RMQ

### ISSUE THREAD

**Issue by xtreme-sameer-vohra** [knative-sandbox/eventing-rabbitmq#645]
This covers
- [ ]  Broker Ingress
- [x]  Trigger Dispatcher #671

**Comment by gab-satchi** [comment]
Moving this out of MVP as ingress timeouts aren't currently part of the eventing spec. Will work on driving the change through core eventing before implementing here.

**Comment by embano1** [comment]
Just checked the code (please correct me if I'm wrong) so that anyone interested in this topic has a better understanding of the connection handling from the `ingress` to the RMQ Cluster (via `amqp091-go`).

> **Note** Evaluation done on the following commit of this repo:
> ```
> git rev-parse head
> dfd65c5bec16c4758218b67fe2095f2055fdfdeb
> ```

It seems for establishing the initial `net.Conn` to RMQ, the `amqp091-go` defaults are used:

https://github.com/knative-sandbox/eventing-rabbitmq/blob/dfd65c5bec16c4758218b67fe2095f2055fdfdeb/vendor/github.com/rabbitmq/amqp091-go/connection.go#L25

The connection is periodically probed/checked via the `Heartbeat` interval, which also seems to be using the default:

https://github.com/knative-sandbox/eventing-rabbitmq/blob/dfd65c5bec16c4758218b67fe2095f2055fdfdeb/vendor/github.com/rabbitmq/amqp091-go/connection.go#L24

**tl;dr: During network/server issues, those affect any publisher, i.e. `ingress`.**

Another issue IMHO is a forever [blocking](https://github.com/rabbitmq/amqp091-go/issues/92) (deadlocking?) event publishing call:

https://github.com/knative-sandbox/eventing-rabbitmq/blob/dfd65c5bec16c4758218b67fe2095f2055fdfdeb/cmd/ingress/main.go#L201

Recently, a PR was added to `amqp091-go` which allows passing a `context.Context` when publishing to avoid this scenario: https://github.com/rabbitmq/amqp091-go/pull/96

Might be a good one to implement on pkg bump and using a sane `context` timeout. cc/ @gab-satchi @salaboy
